src/plugins/playlist/playlist.js

```diff
@@ -0,0 +1,128 @@
+import { sliceProperties } from 'utils/slicing';
+import { normalizeJsonResponse } from 'utils/api';
+import { assign } from 'utils/assign';
+import { PLAYER_EVENT } from 'utils/consts';
+import { isPlainObject } from 'utils/type-inference';
+import { extendCloudinaryConfig, getCloudinaryUrl } from 'plugins/cloudinary/common';
+
+import Playlist from './ui/playlist';
+import PlaylistWidget from './ui/playlist-widget';
+import './ui/panel/playlist-panel';
+
+const LIST_BY_TAG_PARAMS = { format: 'json', resource_type: 'video', type: 'list' };
+
+const playlist = (player, options = {}) => {
+  const _chainTarget = sliceProperties(options, 'chainTarget').chainTarget;
+  let _playlist = null;
+  let _playlistDisposer = null;
+  let _playlistWidget = null;
+
+  const disposePlaylist = () => {
+    player.removeClass('vjs-playlist');
+    const playlist = player.playlist();
+    _playlist = undefined;
+    playlist.dispose();
+    player.off('cldsourcechanged', _playlistDisposer);
+  };
+
+  const createPlaylist = (sources, options) => {
+    if (sources instanceof Playlist) {
+      _playlist = sources;
+      _playlist.resetState();
+      _playlist.currentIndex(_playlist.currentIndex());
```

Comment:
Why do you say that? that line was [always there](https://github.com/cloudinary/cloudinary-video-player/blob/master/src/mixins/playlistable.js#L45)
Looks like it's intended for resetting the playlist index, if a playlist was updated or re-created

Reply:
buy you are getting the index from `_playlist` and setting it to `_playlist`, so it is like `x.setCurrentIndex(x.getCurrentIndex())`

Reply:
not really, the way I read it is - if this playlist is already initialized, skip to the `currentIndex` it was on, otherwise go back to 0
but I'm honestly not sure when that `if` will be truthy
<img width="509" alt="image" src="https://github.com/cloudinary/cloudinary-video-player/assets/346835/e046d4af-6137-4df5-96e0-e7f8ea7f9aaa">